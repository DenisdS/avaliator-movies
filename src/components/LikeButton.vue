<script>
export default {
  name: 'LikeButton',
  data() {
    return {
      title: ''
    }
  }
}
</script>

<template>
  <form @click.prevent="$emit('liked', title)">
    <input type="text" v-model="title" />
    <button type="button">Enviar</button>
  </form>
</template>