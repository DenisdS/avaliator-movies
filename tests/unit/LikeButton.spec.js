import { shallowMount } from '@vue/test-utils'
import LikeButton from '@/components/LikeButton'

describe('LikeButton', () => {
  const build = () => {
    const wrapper = shallowMount(LikeButton)

    return {
      wrapper,
      input: () => wrapper.find('input'),
      button: () => wrapper.find('button'),
    }
  }

  it('renders the component', () => {

    const { wrapper } = build()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {

    const { button } = build()

    expect(button().exists()).toBe(true)
  })

  it('calls "liked" event when user click "like button"', () => {
    
    const expectedCard = 'The Matrix'
    const { wrapper, input, button } = build()
    input().element.value = expectedCard

   input().trigger('input')
   button().trigger('click')

   expect(wrapper.emitted().liked[0]).toEqual([expectedCard])
  })
})