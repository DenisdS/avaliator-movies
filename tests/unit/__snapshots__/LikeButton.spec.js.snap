// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButton renders the component 1`] = `<form><input type="text"> <button type="button">Enviar</button></form>`;
